### PR TITLE
fix: clevertap sdk version to latest, check mandatory traits for identify

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## PR Description

Currently our Clevertap JS integration uses `a.js` which is 2 years old. Starting from April 21, Clevertap released v1.0.0 out of `a.js`. We are updating the version and making necessary checks in traits for identify calls as per the new version.

## Notion ticket

Ticket link

## Screenshots

Please add screenshots for any new features or UI bug fixes for the following browsers -

- Chrome
  >
- Firefox
  >
- Safari
  >

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: src/integrations/Clevertap/browser.js [hunk 1]
```diff
@@ -60,7 +62,12 @@ class Clevertap {
       onUserLogin: [],
       notifications: [],
     };
+    const { CLEVERTAP } = this.analytics.loadOnlyIntegrations;
+    this.optOut = CLEVERTAP.optOut || false;
+    this.useIP = CLEVERTAP.useIP || false;
     window.clevertap.enablePersonalization = true;
+    window.clevertap.privacy.push({ optOut: this.optOut }); // set the flag to true, if the user of the device opts out of sharing their data
+    window.clevertap.privacy.push({ useIP: this.useIP }); // set the flag to true, if the user agrees to share their IP data
```
Comment (ItsSudip) [inline src/integrations/Clevertap/browser.js:70]:
```suggestion
    window.clevertap.privacy.push({ optOut: this.optOut }); 
    window.clevertap.privacy.push({ useIP: this.useIP });
```
These comments are not required rather you can share the SDK doc of clevertap.

Comment (yashasvibajpai) [reply src/integrations/Clevertap/browser.js]: updated

**Comment by krishna2020** [comment]
Bundle size check failed @yashasvibajpai

File: .size-limit.js [hunk 2]
```diff
@@ -13,7 +13,7 @@ module.exports = [
     name: 'All Integrations - CDN',
     path: 'dist/legacy/js-integrations/*.min.js',
     gzip: true,
-    limit: '432.6 kB',
+    limit: '432.654 kB',
```
Comment (MoumitaM) [inline .size-limit.js:16]:
```suggestion
    limit: '432.7 kB',
```

Comment (MoumitaM) [reply .size-limit.js:16]: No need to give such precise value.

**Review by MoumitaM — CHANGES_REQUESTED** [review]

File: src/integrations/Clevertap/browser.js [hunk 3]
```diff
@@ -48,19 +48,32 @@ class Clevertap {
...
     logger.debug('===in init Clevertap===');
+    // START-NO-SONAR-SCAN
     const sourceUrl =
-      document.location.protocol == 'https:'
-        ? 'https://d2r1yp2w7bby2u.cloudfront.net/js/a.js'
-        : 'http://static.clevertap.com/js/a.js';
+      document.location.protocol === 'https:'
+        ? 'https://d2r1yp2w7bby2u.cloudfront.net/js/clevertap.min.js'
+        : 'http://static.clevertap.com/js/clevertap.min.js';
+    // END-NO-SONAR-SCAN
```
Comment (saikumarrs) [inline src/integrations/Clevertap/browser.js:56]: Why is the scanning turned off for just one file?

Comment (yashasvibajpai) [reply src/integrations/Clevertap/browser.js:56]: Sonar was not allowing http endpoint of clevertap so turned off scanning for the urls.

File: .size-limit.js (hunk 2 above)
Comment (yashasvibajpai) [reply .size-limit.js:16]: Ok sure, will update the size

File: src/integrations/Clevertap/browser.js (hunk 3 above)
Comment (saikumarrs) [reply src/integrations/Clevertap/browser.js:56]: Got it. You might want to add an inline comment to clarify that.

**Review by MoumitaM — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**